### DISCUSSION TITLE
关于Prism使用的几个问题

### DISCUSSION THREAD

**Discussion by CoastRed** [BYJRK/DotNet-Discussions#39] [MVVM]
1.  在prism使用时，如果我需要在ViewModel中直接访问View中的某个控件，（比如获取某个控件的实际高度之类的），有什么优雅的办法？View和ViewModel我是在app.xaml.cs中用的以下方式绑定的，用的容器
```C#
containerRegistry.RegisterDialog<SearchMedView, SearchMedViewModel>();
```
2. 在prism中怎么像asp .net core中那样进行快速服务的注册
比如在asp .net core中注入内存缓存可以使用扩展方法
```C#
builder.Services.AddMemoryCache()
``` 
在prism中有没有这种使用扩展方法的类似方式，注意是扩展方法，不是下面这种还需要自己先new一个的方式
```C#
containerRegistry.RegisterSingleton<ILogger>(() =>
            {
                return this.logger;
            });
```

3. 在使用按钮绑定命令时，使用CommunityToolkit.Mvvm包时，如果是异步命令，在命令没有执行完时，按钮会自动灰色，不允许再次点击，命令执行完成后会自动有允许点击，并且view中还能获取命令的状态，以及命令可以自动和token集成，请问在使用prism时，prism中是否有这种方式，还是需要自己实现？

**Comment by BYJRK** [comment]
1. 为什么你会有这种需求？如果你要做的事情是前台效果，那么你应该考虑完全放在前台写。如果真有和 ViewModel 联动的需要，可以考虑将控件的高度 OneWayToSource 绑定到 ViewModel 的属性上，或者用 Messenger 也是可行的
2. 如果想在 WPF 中使用 ASP.NET 的 service，可以参考一下 SingletonSean 的视频，比如 https://www.youtube.com/watch?v=3EzHn9ir5M8
3. 我并不太熟悉 Prism，不过你可以参考一下 CommunityToolkit.Mvvm 中 [AsyncRelayCommand 的实现](https://github.com/CommunityToolkit/dotnet/blob/ee0587ebf9bb060edf38799b44098dae98ec5314/src/CommunityToolkit.Mvvm/Input/AsyncRelayCommand.cs#L23)，了解一下它是怎样在内部记录异步任务的状态，以及通知 CanExecute 的

**Reply by sw554227643** [reply]
1、创建一个IView，这个接口里面写你需要的东西
2、将View继承这个IView 
3、在view的.cs里面将这个view传到 viewmod里面
4、
 public virtual void Init(IView view)
        {
            _view = view;
            _view.DataContext = this;
            _view.Loaded += _view_Loaded;
        }
就行了
